src/status/libstatus/settings.nim

```diff
@@ -24,14 +24,16 @@ proc getSettings*(useCached: bool = true): JsonNode =
   dirty = false
   result = settings
 
-proc getSetting*[T](name: string, defaultValue: T, useCached: bool = true): T =
+proc getSetting*[T](name: Setting, defaultValue: T, useCached: bool = true): T =
   let settings: JsonNode = getSettings(useCached)
-  if not settings.contains(name):
+  if not settings.contains($name) or $settings{$name} == "{}":
     return defaultValue
-  result = Json.decode($settings{name}, T)
+  result = Json.decode($settings{$name}, T)
 
-proc getSetting*[T](name: string, useCached: bool = true): T =
-  let settings: JsonNode = getSettings(useCached)
-  if not settings.contains(name):
-    return default(type(T))
-  result = Json.decode($settings{name}, T)
+proc getSetting*[T](name: Setting, useCached: bool = true): T =
+  result = getSetting(name, default(type(T)), useCached)
+
+proc getCurrentNetwork*(): Network =
+  result = Network.Mainnet
+  if getSetting[string](Setting.Networks_CurrentNetwork, constants.DEFAULT_NETWORK_NAME) == "testnet_rpc":
+    result = Network.Testnet
```

Comment:
We only support two networks for now? Should we make this a `switch` instead?

Reply:
The networks should have an attribute to obtain the type of network. In the app it's possible to add custom networks, and it asks if the network being added is Mainnet/Ropsten/Rinkeby/Custom

Reply:
Yes, fair, but I didn't want to go crazy with this PR and just wanted to get Ropsten support since we have a faucet on Ropsten and can use it to get STT so we can interact with contracts.